src/libs/merge/kdbmerge.c

```diff
@@ -609,6 +648,223 @@ static int checkSingleSet (KeySet * checkedSet, KeySet * firstCompared, KeySet *
 	return 0;
 }
 
+#ifdef LIBGITFOUND
+
+/**
+ * Iterates over all keys that belong to an key array started by arrayStart and puts their values into a newly allocated string
+ * Each value is separated by a newline.
+ * Does not modify arrayStart.
+ *
+ * @param ks the key set of which the array is part
+ * @param arrayStart key that is the start of the array /#0
+ * @param informationKey the key for error codes
+ * @returns pointer to the string, NULL on error
+ * */
+static char * getValuesAsArray (KeySet * ks, const Key * arrayStart, Key * informationKey)
+{
+	if (ks == NULL || arrayStart == NULL || informationKey == NULL)
+	{
+		ELEKTRA_SET_INTERNAL_ERROR (informationKey, "Parameters must not be null");
+		return NULL;
+	}
+	int size = 1; // size 1 for final \0
+	/**
+	 * this calloc sets the first character to \0 so that the first invocation of
+	 * strncat has something to append to
+	 */
+	char * result = elektraCalloc (1);
+	/** ksLookup does not get a copy of the key but the real key.
+	 *  The elektraArrayIncName would then change the name of the real key.
+	 *  We don't want that as we only increase the name to loop over all keys.
+	 */
+	Key * iterator = keyDup (arrayStart);
+	if (iterator == NULL)
+	{
+		ELEKTRA_SET_INTERNAL_ERROR (informationKey, "Could not duplicate key to iterate.");
+		elektraFree (result);
+		return NULL;
+	}
+	Key * lookup;
+	int counter = 0;
+	while ((lookup = ksLookup (ks, iterator, KDB_O_POP)) != 0)
+	{
+		counter++;
+		int tmpSize = keyGetValueSize (lookup);
+		size += tmpSize; // keyGetValueSize includes size for \0 which we use for \n instead
+		if (elektraRealloc ((void **) &result, size) < 0)
```

Comment:
only realloc if size is bigger than what was already allocated.

Reply:
> only realloc if size is bigger than what was already allocated.

This was always the case. I now implemented another solution that might come closer to what you expected.